### PR TITLE
parsing_new_package_omer

### PR DESCRIPTION
**Pull request by omerilbilgi**

### CODE REVIEW

**Comment by omerilbilgi** [comment]
parsing_new_package_omer

File: communication.py [hunk 1]
```diff
@@ -75,73 +75,83 @@ def pckParser(self, line):
         # header = line[0] + line[1]
         length = line[2]
         length = int.from_bytes(length, "little", signed=False)
-
-        takim_no = line[3] + line[4]
-        takim_no = int.from_bytes(takim_no, "little", signed=False)
-        takim_no = takim_no * 8
-
-        paket_no = line[5] + line[6]
+        paket_no = line[4] + line[5]
```
Comment (fthrslntgy) [inline communication.py:78]: where is third byte? all of the numbers below here are written as one more than they should be. fix it.

File: communication.py [hunk 2]
```diff
@@ -75,73 +75,83 @@ def pckParser(self, line):
...
+        takim_no = line[70] + line[71]
+        takim_no = int.from_bytes(takim_no, "little", signed=False)
+        takim_no = takim_no * 8
+
+        finish = line[72] + line[73]
+        finish = int.from_bytes(finish, "little", signed=False)
+
+        crc = line[74]
+        crc = int.from_bytes(crc, "little", signed=False)
+    #    return_number = line[82]
```
Comment (fthrslntgy) [inline communication.py:140]: remove all old lines. these are unusable for this year. delete all.

File: communication.py [hunk 3]
```diff
@@ -163,16 +173,16 @@ def pckParser(self, line):
         row.append(latitude_pl)
         row.append(longitude_pl)
         row.append(float("{:.2f}".format(altitude_pl)))
-        row.append(latitude_car)
-        row.append(longitude_car)
-        row.append(float("{:.2f}".format(altitude_car)))
+    #    row.append(latitude_car)
```
Comment (fthrslntgy) [inline communication.py:176]: remove all old lines at here (row.append part)

File: communication.py [hunk 4]
```diff
@@ -163,16 +173,16 @@ def pckParser(self, line):
...
         row.append(float("{:.2f}".format(yaw)))
         row.append(float("{:.2f}".format(roll)))
         row.append(float("{:.2f}".format(pitch)))
-        row.append(return_number)
-        row.append(video_status)
-        row.append(weather_forecast)
+    #    row.append(return_number)
+    #    row.append(video_status)
+    #    row.append(weather_forecast)
         # print(row)
```
Comment (fthrslntgy) [inline communication.py:186]: you can uncomment printing row.

File: communication.py [hunk 5]
```diff
@@ -187,7 +197,7 @@ def pckParser(self, line):
         self.widget.graphs.update_sp_tmp_v(abs(speed), tempe, b_voltage)
```
Comment (fthrslntgy) [inline communication.py:197]: all lines below here should be in comment. this part is unusable for now. you should check that is your parsing valid. visualize data later.

File: constants.py [hunk 6]
```diff
@@ -3,6 +3,7 @@
 TELEMETRY_LEN = 88
 TELEMETRY_PERIOD = 1
 SAT_STATUS_VARS = ("0. Telekomut yok", "1. Görev Yükü Ayrılma", "2. Görev Yükü Kilitlenme", "3. Uçuşa Hazır" ,"4. Model Uydu İniş" ,"5. Ayrılma", "6. Görev Yükü İniş", "7. Kurtarma")
+COMMAND = ("START", "STOP", "RECORD", "PAUSE", "UNPAUSE", "UNRECORD")
```
Comment (fthrslntgy) [inline constants.py:6]: nice

**Review by fthrslntgy — CHANGES_REQUESTED** [review]

File: communication.py [hunk 7]
```diff
@@ -75,73 +75,83 @@ def pckParser(self, line):
...
-        return_number = line[82]
-        return_number = int.from_bytes(return_number, "little", signed=False)
-        video_status = line[83]
-        video_status = int.from_bytes(video_status, "little", signed=False)
-        weather_forecast = line[84]
-        weather_forecast = int.from_bytes(weather_forecast, "little", signed=False)
+
+        takim_no = line[70] + line[71]
+        takim_no = int.from_bytes(takim_no, "little", signed=False)
+        takim_no = takim_no * 8
```
Comment (fthrslntgy) [inline communication.py:133]: this part is written for team number at last year. that number was more bigger than 2 bytes and after received that, we were multiplying it with 8. (so it was sended as divided by 8). this part is not necessary for now.

**Comment by fthrslntgy** [comment]
merge conflict since oruc's changes. cant merged but not invalid.

**Pull request closed without merging**